pkg/reconciler/builder/builder.go

```diff
@@ -176,7 +185,30 @@ func (c *Reconciler) reconcileBuilder(ctx context.Context, builder *buildapi.Bui
 		return buildapi.BuilderRecord{}, err
 	}
 
-	return c.BuilderCreator.CreateBuilder(keychain, clusterStore, clusterStack, builder.Spec.BuilderSpec)
+	fetcher := cnb.NewRemoteBuildpackFetcher(c.KeychainFactory, clusterStore, buildpacks, clusterBuildpacks)
+
+	buildpackSources, buildRecord, err := c.BuilderCreator.CreateBuilder(ctx, keychain, fetcher, clusterStack, builder.Spec.BuilderSpec)
+	if err != nil {
+		return buildapi.BuilderRecord{}, err
+	}
+
+	for _, source := range buildpackSources {
+		err = c.Tracker.Track(reconciler.Key{
```

Comment:
Is skipping tracking unused resources a problem if this leads to them not trigger updates to builders if they would be used by the builders?

Reply:
I'm not quite sure I'm following what you mean by "if they would be used by the builders". Are you saying if a new version of a meta-buildpack references a previously untracked buildpack?

Reply:
Something like:

1. We have 3 Buildpack CRs, `paketo-buildpacks/go`, `paketo-buildpacks/go-dist`, and `paketo-buildpacks/go-vendor`
1. `paketo-buildpacks/go:1.0.0` is a meta-buildpack and only uses `go-dist`
1. `paketo-buildpacks/go:2.0.0` is a meta-buildpack and uses both `go-dist` and `go-vendor`
1. We create a Builder using `paketo-buildpacks/go:1.0.0`
1. The reconciler starts tracking `go` and `go-dist`
1. We bump the builder to `paketo-buildpacks/go:2.0.0`
1. It'll reconcile the entire builder again and figure out it needs to track all 3 Buildpacks now

Reply:
Talked offline, we need to track all Builpack resource changes because if a Builder uses only the `id` field, it's possible for a new Buildpack CR to be created with a newer version

Reply:
fixed in b7e42f1efd4329ee7794a88cd310ba13f0184b0d